### PR TITLE
assert/raise modifications

### LINKED ISSUE #1902: VIP: Allow raise without reason string
## Simple Summary
Allow use of `raise` without a reason string.

## Motivation
1. I see no technical reason to enforce requiring a revert string. A bare `raise` statement is parseable.
2. Requiring a reason for `raise` but not `assert` is inconsistent.
3. The requirement is easily circumvented:
```python
assert False
```

## Specification
Fairly self-explanatory. The following syntax should compile:

```python
if cond:
    raise
```

## Backwards Compatibility
Will not cause any issues.

## Copyright
Copyright and related rights waived via [CC0](https://creativecommons.org/publicdomain/zero/1.0/)

### PR DESCRIPTION
**Pull request by iamdefinitelyahuman**
### What I did
* remove constancy checks on `assert` statements
* remove the `assert_modifiable` builtin function
* allow `raise` without a reason, with a reason >32 chars long, or with `UNREACHABLE`

The documentation isn't perfect, but I plan to add a new section for statements where I'll end up rewriting all of this anyway.

Closes #1902 

### How to verify it
Run tests.

### Cute Animal Picture
image

### CODE REVIEW

File: vyper/parser/context.py [hunk 1]
```diff
@@ -94,13 +94,6 @@ def assignment_scope(self):
         yield
         self.in_assignment = False
 
-    @contextlib.contextmanager
-    def assertion_scope(self):
```
Comment (fubuloubu) [inline vyper/parser/context.py:98]: what was this?

File: vyper/parser/expr.py [hunk 2]
```diff
@@ -653,12 +653,6 @@ def load_bytearray(side):
     def parse_BoolOp(self):
         # Iterate through values
         for value in self.expr.values:
-            # Check for calls at assignment
-            if self.context.in_assignment and isinstance(value, vy_ast.Call):
```
Comment (fubuloubu) [inline vyper/parser/expr.py:657]: this was just to handle assert basically?

File: vyper/parser/external_call.py [hunk 3]
```diff
@@ -37,8 +37,7 @@ def external_call(node, context, interface_name, contract_address, pos, value=No
     ]
     if context.is_constant() and not sig.const:
         raise ConstancyViolation(
```
Comment (fubuloubu) [inline vyper/parser/external_call.py:39]: Is the context usage changed now where this is a broader error?

File: vyper/parser/stmt.py [hunk 4]
```diff
@@ -194,14 +190,19 @@ def _assert_reason(self, test_expr, msg):
         return LLLnode.from_list(assert_reason, typ=None, pos=getpos(self.stmt))
 
     def parse_Assert(self):
-        with self.context.assertion_scope():
```
Comment (fubuloubu) [inline vyper/parser/stmt.py:197]: This was the assert_modifiable scope, right?

Comment (iamdefinitelyahuman) [reply vyper/parser/stmt.py:197]: Yep

File: vyper/parser/expr.py (hunk 2 above)
Comment (iamdefinitelyahuman) [reply vyper/parser/expr.py:657]: Yeah

File: vyper/parser/external_call.py (hunk 3 above)
Comment (iamdefinitelyahuman) [reply vyper/parser/external_call.py:39]: Hrmm..  I thought it was still necessary, but it shows as not covered on codecov so probably can go.

**Pull request merged**